packages/sdk/src/types/Vouchers.ts

```diff
@@ -0,0 +1,206 @@
+import { OrdersGetResponse } from './Orders'
+import { SimpleCustomer } from './Customers'
+
+interface DiscountUnit {
+	type?: 'UNIT'
+	unit_off?: number
+	effect?: string
+}
+
+interface DiscountAmount {
+	type?: 'AMOUNT'
+	amount_off?: string
+}
+
+interface DiscountPercent {
+	type?: 'PERCENT'
+	percent_off?: number
+}
+
+export interface VouchersResponse {
+	id: string
+	code: string
+	campaign?: string
+	category?: string
+	type?: 'DISCOUNT_VOUCHER' | 'GIFT_VOUCHER'
+	discount: DiscountAmount | DiscountPercent | DiscountUnit
+	gift?: {
+		amount: number
+	}
+	loyalty_card?: {
+		points: number
+	}
+	start_date?: string
+	expiration_date?: string
+	validity_timeframe?: {
+		interval: string
+		duration: string
+	}
+	validity_day_of_week?: number[]
+	publish?: {
+		object: 'list'
+		count: number
+		data_ref: 'entries'
+		entries: $FixMe[]
+		total: number
+		url: string
+	}
+	redemption?: {
+		object: 'list'
+		quantity?: number
+		redeemed_quantity: number
+		data_ref: 'redemption_entries'
+		redemption_entries: $FixMe[]
+		total: number
+		url: string
+	}
+	active: boolean
+	additional_info?: string
+	metadata?: Record<string, any>
+	assets?: {
+		qr?: {
+			id: string
+			url: string
+		}
+		barcode?: {
+			id: string
+			url: string
+		}
+	}
+	is_referral_code: boolean
+	referrer_id?: string
+	holder_id?: string
+	updated_at?: string
+	object: 'voucher'
+	validation_rules_assignments: {
+		object: 'list'
+		total: number
+		data_ref: 'data'
+		data?: {
+			id: string
+			rule_id?: string
+			related_object_id?: string
+			related_object_type?: string
+			created_at: string
+			object: 'validation_rules_assignment'
+		}[]
+	}
+}
+
+export interface VouchersQualificationExamineBody {
+	customer?: Omit<SimpleCustomer, 'object'> & { description: string }
+	order?: Pick<OrdersGetResponse, 'id' | 'source_id' | 'amount' | 'items' | 'metadata'>
+	reward?: {
+		id: string
+		assignment_id?: string
+	}
+	metadata?: Record<string, any>
+}
+
+export interface VouchersQualificationExamineParams {
+	audienceRulesOnly?: boolean
+	order?: 'created_at' | '-created_at' | 'updated_at' | '-updated_at'
+	limit?: number
+}
+
+export interface VouchersQualificationExamineResponse {
+	id: string
+	created_at: string
+	object: 'list'
+	total: number
+	data_ref: 'data'
+	data?: VouchersResponse[]
+}
+
+export type VouchersCreate = Pick<
```

Comment:
It seems like it's broken.
I can not use following piece of code to create voucher in `.ts` file:
```
voucherify.vouchers
	.create({
		code: 'SDK 01',
		type: 'GIFT_VOUCHER',
		gift: {
			amount: 1000,
		},
	})
	.then(voucher => {
		console.log('CREATED: %j:', voucher)
	})
	.catch(error => {
		console.error(error)
	})
```
`discount` field should not be required since I want to create just `gift` card voucher - now it requires to pass `discount` field also resulting in 400 Bad request
`active` field should not be required as by default voucher is active

Reply:
`code` is actually also not required since you can use `code_config` to define what code should be generated

Reply:
Additionally now `redemption` requires you to pass a list structure so it is completely wrong for creating voucher.